streams/shared/src/main/scala/zio/stream/ZSink.scala

```diff
@@ -12,76 +12,76 @@ import zio._
 // - Sinks should always end when receiving a `None`. It is a defect to not end with some
 //   sort of result (even a failure) when receiving a `None`.
 // - Sinks can assume they will not be pushed again after emitting a value.
-abstract class ZSink[-R, +E, -I, +Z] private (
-  val push: ZManaged[R, Nothing, ZSink.Push[R, E, I, Z]]
+abstract class ZSink[-R, +E, -I, +Z, +J] private (
```

Comment:
```suggestion
abstract class ZSink[-R, +E, -I, +L, +Z] private (
```
The target of "map" should always be the last type parameter. Moreover, please use something indicative of leftovers: `L`, `X`, etc.